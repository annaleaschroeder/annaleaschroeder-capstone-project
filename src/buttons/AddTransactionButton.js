import React from 'react'
import styled from 'styled-components/macro'

export default function AddTransactionButton() {
  return (
    <ButtonContainer>
      <AddTrxButton>
        <AddTrxRoute href="/add-new-transaction">
          <AddImage
            xmlns="http://www.w3.org/2000/svg"
            height="40"
            viewBox="0 0 24 24"
            width="40"
            fill="white"
          >
            <path d="M0 0h24v24H0V0z" fill="none" />
            <path d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z" />
          </AddImage>
        </AddTrxRoute>
      </AddTrxButton>
    </ButtonContainer>
  )
}

const ButtonContainer = styled.div`
  display: flex;
  height: auto;
  width: auto;
  z-index: 100;
  bottom: 40px;
  justify-content: center;
  justify-self: center;
  justify-items: center;
`

const AddTrxButton = styled.button`
  display: flex;
  border: none;
  border-radius: 50px;
  box-shadow: 5px 5px 10px var(--grey-shadow);
  background: var(--blue-main);
  padding: 20px;
  cursor: pointer;
  position: fixed;
  z-index: 10;
  bottom: 40px;
`
const AddTrxRoute = styled.a`
  text-decoration: none;
  height: auto;
  width: auto;
`

const AddImage = styled.svg`
  display: inline;
  justify-items: center;
`
