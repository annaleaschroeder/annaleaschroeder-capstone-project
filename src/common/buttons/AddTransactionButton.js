import React from 'react'
import styled from 'styled-components/macro'

export default function AddTransactionButton() {
  return (
    <AddTrxRoute href="/add-new-transaction">
      <AddImage
        xmlns="http://www.w3.org/2000/svg"
        height="40"
        viewBox="0 0 24 24"
        width="40"
        fill="white"
      >
        <path d="M0 0h24v24H0V0z" fill="none" />
        <path d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z" />
      </AddImage>
    </AddTrxRoute>
  )
}

const AddTrxRoute = styled.a`
  text-decoration: none;
  display: flex;
  border: none;
  border-radius: 50px;
  box-shadow: 5px 5px 10px var(--grey-shadow);
  background: var(--blue-main);
  padding: 20px;
  cursor: pointer;
  position: fixed;
  z-index: 10;
  bottom: 40px;
`

const AddImage = styled.svg`
  display: inline;
  justify-items: center;
`
