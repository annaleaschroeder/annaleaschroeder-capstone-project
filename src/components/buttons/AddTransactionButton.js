import React from 'react'
import styled from 'styled-components/macro'

export default function AddTransactionButton() {
  return (
    <ButtonContainer>
      <AddTrxRoute href="/add-new-transaction">
        <AddTrxButton>
          <AddImage
            xmlns="http://www.w3.org/2000/svg"
            height="24"
            viewBox="0 0 24 24"
            width="24"
          >
            <path d="M0 0h24v24H0V0z" fill="none" />
            <path d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z" />
          </AddImage>
          <ButtonText>Add Transaction</ButtonText>
        </AddTrxButton>
      </AddTrxRoute>
    </ButtonContainer>
  )
}

const ButtonContainer = styled.div`
  display: flex;
  height: fit-content;
  width: auto;
  justify-content: right;
`

const AddTrxRoute = styled.a`
  text-decoration: none;
`

const AddTrxButton = styled.button`
  display: flex;
  border: none;
  box-shadow: 5px 5px 10px var(--grey-shadow);
  border-radius: 5px;
  cursor: pointer;
`
const ButtonText = styled.span`
  align-items: center;
  padding-top: 4px;
`
const AddImage = styled.svg`
  display: inline;
  justify-items: center;
`
