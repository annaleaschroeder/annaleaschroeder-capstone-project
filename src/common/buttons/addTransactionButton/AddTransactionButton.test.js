import React from 'react'
import AddTransactionButton from './AddTransactionButton'
import { render } from '@testing-library/react'

describe('AddTransactionButton', () => {
  it('renders correctly', () => {
    const { container } = render(<AddTransactionButton />)
    expect(container).toMatchSnapshot()
  })
})
