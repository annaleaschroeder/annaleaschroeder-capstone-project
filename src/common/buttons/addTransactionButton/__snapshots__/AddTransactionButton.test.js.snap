// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTransactionButton renders correctly 1`] = `
<div>
  <a
    class="AddTransactionButton__AddTrxRoute-sc-12jarje-0 bcjktu"
    href="/add-new-transaction"
  >
    <svg
      class="AddTransactionButton__AddImage-sc-12jarje-1 dMOSom"
      fill="var(--white)"
      height="40"
      viewBox="0 0 24 24"
      width="40"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0V0z"
        fill="none"
      />
      <path
        d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
      />
    </svg>
  </a>
</div>
`;
